File: ios/Plugin/PrintVPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(printBPlugin, "printB",
           CAP_PLUGIN_METHOD(echo, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(blPrint, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(testingMessage, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(getNotifications, CAPPluginReturnPromise);
)

File: ios/Plugin/PrintVPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(printBPlugin)
public class printBPlugin: CAPPlugin {
    private let implementation = printB()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
    @objc func blPrint(_ call: CAPPluginCall) {
        let value = "Hola"
        call.resolve([
            "value": implementation.echo(value)
        ])
    }

     @objc func testingMessage(_ call: CAPPluginCall) {
        let value = "Hola"
        call.resolve([
            "value": implementation.echo(value)
        ])
    }

    @objc func getNotifications(_ call: CAPPluginCall) {
        let value = "Hola"
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
}
